search/search_integration_tests/processor_test.cpp

```diff
@@ -2038,7 +2046,6 @@ UNIT_CLASS_TEST(ProcessorTest, Strasse)
     checkNoErrors("abcd strasse", rules);
     checkNoErrors("abcd straße ", rules);
     checkNoErrors("abcd straße", rules);
-    checkNoErrors("abcd ", rules);
     checkNoErrors("abcd", rules);
```

Comment:
Not sure here, but probably sequenced equal calls check for no side effect .. But, most likely, they are useless.

Reply:
I think it's testing the trailing space. Adding a trailing space forces a normal match instead of an `m_prefixToken` prefix match.

But `abcd` doesn't normal-match to `abcdstr` since there are three errors and length-four tokens only allow one error. It works for prefix matching since those accept partial matches.

Reply:
Ah, I see! Well, but it should, because "straße" suffix is generated as a separate token. This test is valid and should work.

Reply:
This test should work, because "strasse" suffix is separately processed in generator. Is this test-case breaks your logic somehow?

Reply:
Hm. Going back to the master branch, there are two flavors of prefix matches, and I only support one right now. I'll fix that.

One case is where the prefix token partially matches the result token, like below. That's supported in my code.
```
  test("фотоателье", "фото", TokenRange(0, 1), NAME_SCORE_PREFIX, 0);
```

The other case is where a non-prefix token matches the start of a query:
```
  test("San Francisco", "San ", TokenRange(0, 1), NAME_SCORE_PREFIX, 0);
```
I'm currently parsing that as a SUBSTRING match, but as a human, PREFIX makes more sense.

Let me think on how to write that without adding too many special cases. I'll have something soon.

Reply:
Well, "фото " should not match "фотоателье". But "abcdstrasse" is an exception because of German language specific.

Generator index "abcdstrasse" as 2 separate tokens like "abcd" and "strasse", so "abcd " query will match the first token.